core/src/databases.rs

```diff
@@ -18,22 +18,50 @@ impl MongoDatabases {
     // All columns except the TABLE_CAT column contain NULLs.
     // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
     // timeout, the query must finish before the timeout or an error is returned.
-    pub fn list_all_catalogs(_client: &MongoConnection, _query_timeout: Option<i32>) -> Self {
-        unimplemented!()
+    pub fn list_all_catalogs(
+        mongo_connection: &MongoConnection,
+        _query_timeout: Option<i32>,
+    ) -> Self {
+        let database_names: Vec<String> = mongo_connection
+            .client
+            .list_database_names(None, None)
+            .unwrap();
+        let mut db_names: Vec<Bson> = Vec::new();
+        for name in database_names {
+            db_names.push(Bson::String(name));
+        }
+        MongoDatabases {
+            database_names: db_names,
+            current_db_index: 0,
+        }
     }
 }
 
 impl MongoStatement for MongoDatabases {
     // Increment current_db_index.
     // Return true if current_db_index index is < for databases_names.length.
     fn next(&mut self) -> Result<bool> {
-        unimplemented!()
+        let current = self.current_db_index;
+        self.current_db_index += 1;
+        Ok(current < u32::try_from(self.database_names.len()).unwrap())
     }
 
     // Get the BSON value for the value at the given colIndex on the current row.
-    fn get_value(&self, _col_index: u16) -> Result<Option<&Bson>> {
+    fn get_value(&self, col_index: u16) -> Result<Option<&Bson>> {
         // The mapping for col_index <-> Value will be hard-coded and handled in this function
         // 1-> databases_names[current_row_index]
-        unimplemented!()
+        match col_index {
+            1 => Ok(Some(
+                self.database_names
+                    .get(self.current_db_index as usize)
+                    .unwrap(),
+            )),
+            _ => Ok(Some(&Bson::Null)),
+            // SQL-1031: Add database listing edge case handling
```

Comment:
I would put the logic for validating that a col_index is valid in the sqlTables functions and not here because it will be shared with collection.rs. Also, I think it might be apparent enough in the design doc, but I was looking to keep the core component as ODBC agnostic as we possibly can.
Simply because it is easier to know where these kind of logic is handled if we always follow the same pattern.